src/reflect/value.go

```diff
@@ -35,30 +35,32 @@ func Indirect(v Value) Value {
 	return v.Elem()
 }
 
+//go:linkname compose runtime.compose
+func compose(Type, unsafe.Pointer) interface{}
```

Comment:
I think `compose` and `decompose` are too generic. I would suggest `composeInterface` and `decomposeInterface`, or even `composeInterfaceValue` and `decomposeInterfaceValue`.